packages/client/src/Call.ts

```diff
@@ -1050,6 +1052,9 @@ export class Call {
    */
   private handleSfuSignalClose = (sfuClient: StreamSfuClient) => {
     this.logger('debug', '[Reconnect] SFU signal connection closed');
+    // SFU WS closed before we finished current join, no need to schedule reconnect
+    // because join operation will fail
+    if (this.state.callingState === CallingState.JOINING) return;
```

Comment:
When joining the call, we already wait for the SFU WS connection to be healthy. If that doesn't happen, we shouldn't try reconnecting, and should throw from `join` instead.

When in-call (`callingState === JOINED`) we should still schedule a reconnect.